### PR TITLE
Loadtesting using K6

### LINKED ISSUE #283: implement benchmarks part 1/2
### Description

#281 compliments it 

Implement a tool which will run in our cluster and be able to produce benchmarks.

- simple tool which uses the existing release to derive benchmarks
- at least 5 exit-nodes

What are we testing:

- response time
- request sent delivery rate (success to exit-node)
- response sent delivery rate (success back to entry-node)
- request & response success rate
- matrix of (1, 10, 100, 1000) users (`user-matrix`)

### Relevant issues

- [ ] given a DP endpoint, emulate `user-matrix` and gather metrics

### Notes

- checkout https://github.com/hoprnet/hoprnet/tree/master/tests/k6

### Definition of DONE

When we can run this tool and get benchmarks.

### PR DESCRIPTION
**Pull request by mjadach-iv**
Fixes #283

### CODE REVIEW

File: apps/rpc-server/src/constants.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import * as path from "path";
+require("dotenv").config({ path: path.join(process.cwd(), "./.env.local") });
```
Comment (nionis) [inline apps/rpc-server/src/constants.ts:2]: can you remind me why you needed to add this?

File: devkit/k6-loadtesting/functions/parseHarFile.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+export default function parseHarReq(har) {
```
Comment (nionis) [inline devkit/k6-loadtesting/functions/parseHarFile.js:1]: missing documentation

File: devkit/k6-loadtesting/functions/parseHarReq.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export default function parseHarReq(harReq) {
```
Comment (nionis) [inline devkit/k6-loadtesting/functions/parseHarReq.js:1]: missing documentation

File: devkit/k6-loadtesting/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+## How to test RPC Server
+
+1. Spin up Sandbox 
+2. Spin up RPC Server on http://localhost:3040/
+3. Navigate to `.\devkit\k6-loadtesting\`
+4. Run:
+    - `yarn k6-artificial`  (Artificial usage done with eth_getCode and eth_call)
+    - `yarn k6-blockwallet` (Real usage of blockwallet)
+    - `yarn k6-trustwallet` (Real usage of trustwallet)
+    - `yarn k6-blockwallet`
```
Comment (nionis) [inline devkit/k6-loadtesting/README.md:18]: written twice?

**Review by nionis — CHANGES_REQUESTED** [review]
1. can you update `k6-loadtesting` README that explains how the architecture of the benchmarks work?

2. as the use of this is primarily benchmarks, I suggest to rename from `k6-loadtesting` to `benchmarks`

**Comment by nionis** [comment]
one small thing lets try to include the following pattern on the npm scripts:

```
    "build": "exit 0", // keep exit 0 if none exists
    "dev": "exit 0",  // keep exit 0 if none exists
    "format": "exit 0",  // keep exit 0 if no JS code exists
    "lint": "exit 0",  // keep exit 0 if no JS code exists
    "test": "exit 0",  // keep exit 0 if no tests are needed
    "start:example-1": "...",
    "start:example-2": "...",
    ```

File: apps/rpc-server/src/constants.ts (hunk 1 above)
Comment (mjadach-iv) [reply apps/rpc-server/src/constants.ts:2]: It's needed to start easily the rpc-server without defining the 3 variables every time/
Removing it for now, as it didn't want to work for me.

File: devkit/k6-loadtesting/README.md (hunk 4 above)
Comment (mjadach-iv) [reply devkit/k6-loadtesting/README.md]: @nionis, fixed

File: devkit/k6-loadtesting/functions/parseHarReq.js (hunk 3 above)
Comment (mjadach-iv) [reply devkit/k6-loadtesting/functions/parseHarReq.js]: @nionis, added

**Pull request merged**